tmail-backend/mailbox/opensearch/src/main/java/com/linagora/tmail/mailbox/opensearch/TmailMailboxMappingFactory.java

```diff
@@ -0,0 +1,305 @@
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ *                                                                  *
+ *  This file was taken and adapted from the Apache James project.  *
+ *                                                                  *
+ *  https://james.apache.org                                        *
+ *                                                                  *
+ *  It was originally licensed under the Apache V2 license.         *
+ *                                                                  *
+ *  http://www.apache.org/licenses/LICENSE-2.0                      *
+ ********************************************************************/
+
+/**
+ * This class is copied & adapted from {@link org.apache.james.mailbox.opensearch.DefaultMailboxMappingFactory}
+ */
+
+package com.linagora.tmail.mailbox.opensearch;
+
+import static org.apache.james.backends.opensearch.IndexCreationFactory.CASE_INSENSITIVE;
+import static org.apache.james.backends.opensearch.IndexCreationFactory.KEEP_MAIL_AND_URL;
+import static org.apache.james.backends.opensearch.IndexCreationFactory.KEYWORD;
+import static org.apache.james.backends.opensearch.IndexCreationFactory.RAW;
+
+import java.util.Map;
+
+import org.apache.james.mailbox.opensearch.MailboxMappingFactory;
+import org.apache.james.mailbox.opensearch.json.JsonMessageConstants;
+import org.opensearch.client.opensearch._types.mapping.BooleanProperty;
+import org.opensearch.client.opensearch._types.mapping.DateProperty;
+import org.opensearch.client.opensearch._types.mapping.DynamicMapping;
+import org.opensearch.client.opensearch._types.mapping.KeywordProperty;
+import org.opensearch.client.opensearch._types.mapping.LongNumberProperty;
+import org.opensearch.client.opensearch._types.mapping.NestedProperty;
+import org.opensearch.client.opensearch._types.mapping.ObjectProperty;
+import org.opensearch.client.opensearch._types.mapping.Property;
+import org.opensearch.client.opensearch._types.mapping.RoutingField;
+import org.opensearch.client.opensearch._types.mapping.TextProperty;
+import org.opensearch.client.opensearch._types.mapping.TypeMapping;
+
+import com.google.common.collect.ImmutableMap;
+
+public class TmailMailboxMappingFactory implements MailboxMappingFactory {
```

Comment:
Issue spotted: we would need to be able to specify IndexCreationCustomTokenizer + IndexCreationCustomAnalyzer on James side too. (Defaulting to defaults for now)

Reply:
https://github.com/linagora/tmail-backend/issues/1775#issuecomment-2938866793

I came to the exact same conclusion more or less at the same time

Reply:
All of that is declared in IndexSettings btw so can be just one method